ui/packages/app/src/api.ts

```diff
@@ -31,10 +31,15 @@ interface Payload {
 declare let BUILD_MODE: string;
 declare let BACKEND_URL: string;
 
-const WS_ENDPOINT =
+const WS_PROTOCOL = location.protocol === "https:" ? "wss:" : "ws:";
+const WS_HOST =
 	BUILD_MODE === "dev" || location.origin === "http://localhost:3000"
-		? `ws://${BACKEND_URL.replace("http://", "")}queue/join`
-		: `ws://${location.host}/queue/join`;
+		? BACKEND_URL.replace("http://", "").slice(0, -1)
+		: location.host;
+const WS_PATH = `${
+	location.pathname === "/" ? "" : location.pathname
+}/queue/join`;
+const WS_ENDPOINT = `${WS_PROTOCOL}//${WS_HOST}${WS_PATH}`;
```

Comment:
LoL this feels gibberish to me, it seems I need to study Javascript 🗡️ 

So basically when it is a local host or spaces we will follow this pattern
1. ws://localhost:port/queue-join
2. wss://spaces.huggingface.tech/farukozderim/test-new-queue/queue/join